faunadb-java/src/main/java/com/faunadb/client/types/Ref.java

```diff
@@ -2,60 +2,42 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Optional;
 
 /**
- * A FaunaDB ref type. Create an instance using the {@link Ref#create} method.
- *
- * <p><i>Example</i>: <code>{ "@ref": "classes/spells/102462014988746752" }</code>
- *
- * <p><i>Reference</i>: <a href="https://faunadb.com/documentation/queries#values-special_types">FaunaDB Special Types</a>
+ * A FaunaDB ref type.
  *
+ * @see <a href="https://faunadb.com/documentation/queries#values-special_types">FaunaDB Special Types</a>
  */
-public class Ref extends Value.ConcreteValue {
-  /**
-   * Creates a new {@link Ref} instance.
-   * @param value the value of the Ref.
-   */
-  public static Ref create(String value) {
-    return new Ref(value);
-  }
+public class Ref extends Value.ScalarValue<Value> {
 
-  @JsonProperty("@ref")
-  private final String value;
-
-  @Override
-  public Ref asRef() {
-    return this;
+  @JsonCreator
+  public Ref(@JsonProperty("@ref") String value) {
+    this(new StringV(value));
   }
 
-  @JsonCreator
-  Ref(@JsonProperty("@ref") String value) {
-    this.value = value;
+  public Ref(Value ref) {
```

Comment:
Well, *eventually* it will be possible... either way, `@ref` with anything other than a string literal will fail.